examples/notebooks/beam-ml/run_inference_windowing.ipynb

```diff
@@ -0,0 +1,477 @@
+        "# software distributed under the License is distributed on an\n",
+        "# \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+        "# KIND, either express or implied. See the License for the\n",
+        "# specific language governing permissions and limitations\n",
+        "# under the License"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Apache Beam RunInference Windowing Example\n",
+        "\n",
+        "<table align=\"left\">\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_windowing.ipynb\"><img src=\"https://raw.githubusercontent.com/google/or-tools/main/tools/colab_32px.png\" />Run in Google Colab</a>\n",
+        "  </td>\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://github.com/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_windowing.ipynb\"><img src=\"https://raw.githubusercontent.com/google/or-tools/main/tools/github_32px.png\" />View source on GitHub</a>\n",
+        "  </td>\n",
+        "</table>\n"
+      ],
+      "metadata": {
+        "id": "fKxfINuCPsh9"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "This notebook demonstrates the use of the RunInference transform together with windowing in a streaming pipeline. The pipeline predicts the quality of milk samples and classifies them as 'good', 'bad' or 'medium'. The predictions for each window are then aggregated. The pipeline makes use of the XGBoost model handler. For more information about the RunInference API, see the [Machine Learning section of the Apache Beam documentation](https://beam.apache.org/documentation/ml/overview/).\n",
+        "\n",
+        "With RunInference, a model handlers manages batching, vectorization, and prediction optimization for your XGBoost pipeline or model.\n",
+        "\n",
+        "This notebook demonstrates the following common RunInference patterns:\n",
+        "\n",
+        "- Generate predictions for all samples in a window\n",
+        "- Aggregate the results per window after RunInference\n",
+        "- Print the aggregations"
+      ],
+      "metadata": {
+        "id": "knGVsVR6P_nZ"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Before you begin\n",
+        "Complete the following setup steps:\n",
+        "- Install dependencies for Apache Beam.\n",
+        "- Install XGBoost.\n",
+        "- Download the [Milk Quality Dataset from Kaggle](https://www.kaggle.com/datasets/cpluzshrijayan/milkquality) and put in the current directory with the name `milk_quality.csv` (The dataset should be in csv format)"
+      ],
+      "metadata": {
+        "id": "s5PPNo9HRRe1"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!pip install git+https://github.com/apache/beam.git\n",
```

Comment:
Looks like this line isn't working. Since Beam 2.47 is almost ready to release with the XGBoost changes, I'd probably rather just wait on that and update at the time. We can get this notebook fully approved/ready to go now though so that I just need to hit merge when the time comes